prototype2/common/src/simple_specs.rs

```diff
@@ -24,3 +24,33 @@ macro_rules! spec_systemize {
     }
   }
 }
+
+#[macro_export]
+macro_rules! spec_type {
+  ($i:ident: $([$stuff:tt | $t:ty],)*) => {
+    pub type $i<'a> = ($(spec_duo!{$stuff | $t}),*);
+
+    fn args_from_runarg<'a>(run_arg: &'a specs::RunArg) -> SystemArgs<'a> {
+      run_arg.fetch(|w| {
+        ($(specs_get_expand!{$stuff | w $t}),*)
+      })
+    }
+  }
+}
+
+#[macro_export]
+macro_rules! specs_get_expand {
+  (write | $w:ident $t:ty) => {$w.write_resource::<$t>()};
+  (read | $w:ident $t:ty) => {$w.read_resource::<$t>()};
+  (write_component | $w:ident $t:ty) => {$w.write::<$t>()};
+  (read_component | $w:ident $t:ty) => {$w.read::<$t>()};
+}
+
+
+#[macro_export]
+macro_rules! spec_duo {
+  (write | $t:ty) => {$crate::simple_specs::Write<'a, $t>};
+  (read | $t:ty) => {$crate::simple_specs::Read<'a, $t>};
+  (write_component | $t:ty) => {$crate::simple_specs::AspectWrite<'a, $t>};
+  (read_component | $t:ty) => {$crate::simple_specs::AspectRead<'a, $t>};
```

Comment:
This "'a" is actually unhygenic, but it compiles. I don't know any way to pass a lifetime into a macro, so thats all I got.